ffmpeg 4.0 build failure

1.8.10 fails to build with ffmpeg 4.0

Regressed by https://github.com/FFmpeg/FFmpeg/commit/b79a7da36faa6bb5914b04800b548e99eb3b81ca

```
bash-3.2$ make -j1 V=1
/usr/local/Cellar/cmake/3.11.1/bin/cmake -H/tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10 -B/tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2 --check-build-system CMakeFiles/Makefile.cmake 0
/usr/local/Cellar/cmake/3.11.1/bin/cmake -E cmake_progress_start /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2/CMakeFiles /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2/CMakeFiles/progress.marks
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/Makefile2 all
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/CopyFiles.dir/build.make CMakeFiles/CopyFiles.dir/depend
cd /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2 && /usr/local/Cellar/cmake/3.11.1/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10 /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10 /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2 /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2 /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2/CMakeFiles/CopyFiles.dir/DependInfo.cmake --color=
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/CopyFiles.dir/build.make CMakeFiles/CopyFiles.dir/build
make[2]: Nothing to be done for `CMakeFiles/CopyFiles.dir/build'.
[  0%] Built target CopyFiles
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f src/libutil/CMakeFiles/OpenImageIO_Util.dir/build.make src/libutil/CMakeFiles/OpenImageIO_Util.dir/depend
cd /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2 && /usr/local/Cellar/cmake/3.11.1/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10 /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/src/libutil /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2 /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2/src/libutil /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2/src/libutil/CMakeFiles/OpenImageIO_Util.dir/DependInfo.cmake --color=
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f src/libutil/CMakeFiles/OpenImageIO_Util.dir/build.make src/libutil/CMakeFiles/OpenImageIO_Util.dir/build
make[2]: Nothing to be done for `src/libutil/CMakeFiles/OpenImageIO_Util.dir/build'.
[  8%] Built target OpenImageIO_Util
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f src/libutil/CMakeFiles/hash_test.dir/build.make src/libutil/CMakeFiles/hash_test.dir/depend
cd /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2 && /usr/local/Cellar/cmake/3.11.1/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10 /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/src/libutil /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2 /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2/src/libutil /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2/src/libutil/CMakeFiles/hash_test.dir/DependInfo.cmake --color=
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f src/libutil/CMakeFiles/hash_test.dir/build.make src/libutil/CMakeFiles/hash_test.dir/build
make[2]: Nothing to be done for `src/libutil/CMakeFiles/hash_test.dir/build'.
[  9%] Built target hash_test
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/build.make src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/depend
cd /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2 && /usr/local/Cellar/cmake/3.11.1/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10 /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/src/libOpenImageIO /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2 /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2/src/libOpenImageIO /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2/src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/DependInfo.cmake --color=
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/build.make src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/build
[  9%] Building CXX object src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/__/ffmpeg.imageio/ffmpeginput.cpp.o
cd /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2/src/libOpenImageIO && /usr/local/Homebrew/Library/Homebrew/shims/super/clang++  -DEMBED_PLUGINS=1 -DOpenImageIO_EXPORTS -DUSE_BOOST_ASIO=1 -DUSE_FFMPEG -DUSE_FREETYPE -DUSE_GIF -DUSE_LIBRAW=1 -DUSE_OCIO=1 -DUSE_STD_REGEX -DUSE_WEBP=1 -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -I/tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2/include -I/tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/src/include -isystem /usr/local/include -I/usr/local/include/OpenEXR -I/tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/build-with-python2/src/libOpenImageIO -I/usr/local/include/freetype2  -DNDEBUG -fPIC   -Wall -Werror -fno-math-errno -Wno-unused-function -Wno-overloaded-virtual -Wno-unneeded-internal-declaration -Wno-unused-private-field -Wno-tautological-compare -Qunused-arguments -Wunknown-warning-option -Wno-unused-local-typedefs -std=c++11 -Wno-deprecated-register -UUSE_FIELD3D -Wno-deprecated-declarations -o CMakeFiles/OpenImageIO.dir/__/ffmpeg.imageio/ffmpeginput.cpp.o -c /tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/src/ffmpeg.imageio/ffmpeginput.cpp
/tmp/openimageio-20180421-55436-orqj8w/oiio-Release-1.8.10/src/ffmpeg.imageio/ffmpeginput.cpp:333:72: error: use of undeclared identifier 'CODEC_CAP_DELAY'
    m_codec_cap_delay = (bool) (m_codec_context->codec->capabilities & CODEC_CAP_DELAY);
                                                                       ^
1 error generated.
make[2]: *** [src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/__/ffmpeg.imageio/ffmpeginput.cpp.o] Error 1
make[1]: *** [src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/all] Error 2
make: *** [all] Error 2
bash-3.2$ 
```
https://gist.github.com/ilovezfs/e7786a998d7a18f2011179715f3b7bef

--- a/src/ffmpeg.imageio/ffmpeginput.cpp
+++ b/src/ffmpeg.imageio/ffmpeginput.cpp
@@ -124,6 +124,15 @@ inline int receive_frame(AVCodecContext *avctx, AVFrame *picture,
 }
 #endif
 
+
+// Changes for ffmpeg 4.0
+#define USE_FFMPEG_4_0 (LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(58, 18, 100))
+
+#if USE_FFMPEG_4_0
+#  define CODEC_CAP_DELAY AV_CODEC_CAP_DELAY
+#endif
+
+
 #include <boost/thread/once.hpp>
 
 #include <OpenImageIO/imageio.h>

